### ISSUE TITLE
Any papers connected to this?

### ISSUE THREAD

**Issue by DonaldTsang** [mad-day/sponge-crypto#1]
Would like to see some diagrams about this

**Comment by maxymania** [comment]
No, sorry. This is implementation only. I'm not good at writing papers. And I'm even worse in reading and understanding those.

**Comment by DonaldTsang** [comment]
@maxymania so where did you acquire this algorithm?

**Comment by maxymania** [comment]
At the time I was reading about the [sponge construction](https://en.wikipedia.org/wiki/Sponge_function) and I just wanted to implement it somehow. The only missing part was a Function, so i dropped in the [Core208](https://godoc.org/golang.org/x/crypto/salsa20/salsa#Core208) function of the salsa20 cipher.

The "Pi"-variant has a function, that I read about here: (π-Cipher)[https://aezoo.compute.dtu.dk/doku.php?id=%CF%80-cipher].

**Comment by maxymania** [comment]
The diagram.
![Sponge](https://upload.wikimedia.org/wikipedia/commons/7/70/SpongeConstruction.svg)

**Comment by DonaldTsang** [comment]
@maxymania so you are only using the salsa20 permutation function?
Also could that be replaced by Rabbit's or Sosemanuk's permutation?

**Comment by maxymania** [comment]
Sure.
